src/streaming/kafka_client.rs

```diff
@@ -0,0 +1,115 @@
+use std::{
+    fmt::{Debug, Formatter},
+    time::Duration,
+};
+
+use base64::{engine::general_purpose, Engine as _};
+use eyre::WrapErr;
+use rdkafka::{
+    message::{Header, OwnedHeaders},
+    producer::{FutureProducer, FutureRecord, Producer},
+    ClientConfig,
+};
+
+use super::{KafkaConfig, Message, StreamingClient};
+
+use crate::{Result, Sensitive};
+
+const NO_RETRY: Duration = Duration::from_secs(0);
+
+#[derive(Clone)]
+pub struct KafkaClient {
+    producer: FutureProducer,
+    health_check_topic: String,
+}
+
+impl KafkaClient {
+    /// Creates a Kafka client connected to the broker.
+    ///
+    /// The connection is validated immediately after creation, and if not connected, the
+    /// client creation will fail with an error.
+    pub async fn new(config: &KafkaConfig) -> Result<Self> {
+        tracing::info!(config = ?config, "initing kafka-client");
+
+        // todo: support SSL, broken confirmation and other basic configurations
+        // todo: for now, it is working with a local Kafka in Docker
```

Comment:
```suggestion
        // todo: support SSL, broken confirmation and other basic configurations
        // todo: for now, it is working with a local Kafka in Docker
```

this comment is stale